# Titanic competition with TensorFlow Decision Forests

This notebook will take you through the steps needed to train a baseline Gradient Boosted Trees Model using TensorFlow Decision Forests and creating a submission on the Titanic competition. 

This notebook shows:

1. How to do some basic pre-processing. For example, the passenger names will be tokenized, and ticket names will be splitted in parts.
1. How to train a Gradient Boosted Trees (GBT) with default parameters
1. How to train a GBT with improved default parameters
1. How to tune the parameters of a GBTs
1. How to train and ensemble many GBTs

# Imports dependencies

In [1]:
import numpy as np
import pandas as pd
import os

import tensorflow as tf
import tensorflow_decision_forests as tfdf

print(f"Found TF-DF {tfdf.__version__}")

Found TF-DF 1.2.0


# Load dataset

In [2]:
train_df = pd.read_csv("/kaggle/input/titanic/train.csv")
serving_df = pd.read_csv("/kaggle/input/titanic/test.csv")

train_df.head(10)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

# Prepare dataset

We will apply the following transformations on the dataset.

1. Tokenize the names. For example, "Braund, Mr. Owen Harris" will become ["Braund", "Mr.", "Owen", "Harris"].
2. Extract any prefix in the ticket. For example ticket "STON/O2. 3101282" will become "STON/O2." and 3101282.

In [3]:
def preprocess(df):
    df = df.copy()
    
    def normalize_name(x):
        return " ".join([v.strip(",()[].\"'") for v in x.split(" ")])
    
    # takes the last item of the ticket array; should always give an integer
    def ticket_number(x):
        return x.split(" ")[-1]
    
    # creates a new string from the remaining ticket array
    # concatenates if necessary
    # make sure to not return empty strings, but rather NONE
    def ticket_item(x):
        items = x.split(" ")
        if len(items) == 1:
            return "NONE"
        return "_".join(items[0:-1])
    
    df["Name"] = df["Name"].apply(normalize_name)
    df["Ticket_number"] = df["Ticket"].apply(ticket_number)
    df["Ticket_item"] = df["Ticket"].apply(ticket_item)                     
    return df
    
preprocessed_train_df = preprocess(train_df)
preprocessed_serving_df = preprocess(serving_df)

preprocessed_train_df.head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                              Name     Sex   Age  SibSp  \
0                            Braund Mr Owen Harris    male  22.0      1   
1  Cumings Mrs John Bradley Florence Briggs Thayer  female  38.0      1   
2                             Heikkinen Miss Laina  female  26.0      0   
3         Futrelle Mrs Jacques Heath Lily May Peel  female  35.0      1   
4                           Allen Mr William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked Ticket_number Ticket_item  
0      0         A/5 21171   7.2500   NaN        S         21171         A/5  
1      0          PC 17599  71.2833   C85        C         17599          PC  
2      0  STON/O2. 3101282   7.9250   NaN        S       3101282    STON/O2.  
3      0            113803  53.1000  C123        S        113803        NONE  
4      0            373450   8.0500   NaN        S        373450        NONE

Let's keep the list of the input features of the model. Notably, we don't want to train our model on the "PassengerId" and "Ticket" features.

In [4]:
input_features = list(preprocessed_train_df.columns)
input_features.remove("Ticket")
input_features.remove("PassengerId")
input_features.remove("Survived")
#input_features.remove("Ticket_number")

print(f"Input features: {input_features}")

Input features: ['Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Cabin', 'Embarked', 'Ticket_number', 'Ticket_item']


# Convert Pandas dataset to TensorFlow Dataset

In [5]:
def tokenize_names(features, labels=None):
    """Divide the names into tokens. TF-DF can consume text tokens natively."""
    features["Name"] =  tf.strings.split(features["Name"])
    return features, labels

train_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_train_df,label="Survived").map(tokenize_names)
serving_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_serving_df).map(tokenize_names)

# Train model with default parameters

### Train model

First, we are training a GradientBoostedTreesModel model with the default parameters.

In [6]:
model = tfdf.keras.GradientBoostedTreesModel(
    # let's increase verbosity to get some insights
    verbose=5, # increased to 5
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    random_seed=1234,
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

Use 4 thread(s) for training
Use /tmp/tmp7fom6wac as temporary training directory
Reading training dataset...
Training tensor examples:
Features: {'PassengerId': <tf.Tensor 'data_7:0' shape=(None,) dtype=int64>, 'Pclass': <tf.Tensor 'data_8:0' shape=(None,) dtype=int64>, 'Name': tf.RaggedTensor(values=Tensor("data_4:0", shape=(None,), dtype=string), row_splits=Tensor("data_5:0", shape=(None,), dtype=int64)), 'Sex': <tf.Tensor 'data_9:0' shape=(None,) dtype=string>, 'Age': <tf.Tensor 'data:0' shape=(None,) dtype=float64>, 'SibSp': <tf.Tensor 'data_10:0' shape=(None,) dtype=int64>, 'Parch': <tf.Tensor 'data_6:0' shape=(None,) dtype=int64>, 'Ticket': <tf.Tensor 'data_11:0' shape=(None,) dtype=string>, 'Fare': <tf.Tensor 'data_3:0' shape=(None,) dtype=float64>, 'Cabin': <tf.Tensor 'data_1:0' shape=(None,) dtype=string>, 'Embarked': <tf.Tensor 'data_2:0' shape=(None,) dtype=string>, 'Ticket_number': <tf.Tensor 'data_13:0' shape=(None,) dtype=string>, 'Ticket_item': <tf.Tensor 'data_12:0' sh

[INFO 2024-02-11T15:37:12.777787206+00:00 kernel.cc:756] Start Yggdrasil model training
[INFO 2024-02-11T15:37:12.778862706+00:00 kernel.cc:757] Collect training examples
[INFO 2024-02-11T15:37:12.781899162+00:00 kernel.cc:388] Number of batches: 1
[INFO 2024-02-11T15:37:12.781930909+00:00 kernel.cc:389] Number of examples: 891
[INFO 2024-02-11T15:37:12.783584709+00:00 data_spec_inference.cc:303] 147 item(s) have been pruned (i.e. they are considered out of dictionary) for the column Cabin (0 item(s) left) because min_value_count=5 and max_number_of_unique_values=2000
[INFO 2024-02-11T15:37:12.784316381+00:00 data_spec_inference.cc:303] 1441 item(s) have been pruned (i.e. they are considered out of dictionary) for the column Name (85 item(s) left) because min_value_count=5 and max_number_of_unique_values=2000
[INFO 2024-02-11T15:37:12.784722133+00:00 data_spec_inference.cc:303] 28 item(s) have been pruned (i.e. they are considered out of dictionary) for the column Ticket_item (16 item(

Model trained in 0:00:00.458762
Compiling model...
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: could not get source code
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert
Model compiled.
Accuracy: 0.8260869383811951 Loss:0.8608942627906799


# Train model with improved default parameters

Now you'll use some specific parameters when creating the GBT model

In [7]:
model = tfdf.keras.GradientBoostedTreesModel(
    verbose=5, # Very few logs
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    
    #num_trees=2000,
    
    # Only for GBT.
    # A bit slower, but great to understand the model.
    compute_permutation_variable_importance=True,
    
    # Change the default hyper-parameters
    hyperparameter_template="benchmark_rank1@v1",
    
    #num_trees=1000,
    #tuner=tuner,
    
    min_examples=1,
    categorical_algorithm="RANDOM",
    #max_depth=4,
    shrinkage=0.05,
    #num_candidate_attributes_ratio=0.2,
    split_axis="SPARSE_OBLIQUE",
    sparse_oblique_normalization="MIN_MAX",
    sparse_oblique_num_projections_exponent=2.0,
    num_trees=2000,
    #validation_ratio=0.0,
    random_seed=1234,
    
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

Resolve hyper-parameter template "benchmark_rank1@v1" to "benchmark_rank1@v1" -> {'growing_strategy': 'BEST_FIRST_GLOBAL', 'categorical_algorithm': 'RANDOM', 'split_axis': 'SPARSE_OBLIQUE', 'sparse_oblique_normalization': 'MIN_MAX', 'sparse_oblique_num_projections_exponent': 1.0}.
Use 4 thread(s) for training
Use /tmp/tmpcr6seog2 as temporary training directory
Reading training dataset...
Training tensor examples:
Features: {'PassengerId': <tf.Tensor 'data_7:0' shape=(None,) dtype=int64>, 'Pclass': <tf.Tensor 'data_8:0' shape=(None,) dtype=int64>, 'Name': tf.RaggedTensor(values=Tensor("data_4:0", shape=(None,), dtype=string), row_splits=Tensor("data_5:0", shape=(None,), dtype=int64)), 'Sex': <tf.Tensor 'data_9:0' shape=(None,) dtype=string>, 'Age': <tf.Tensor 'data:0' shape=(None,) dtype=float64>, 'SibSp': <tf.Tensor 'data_10:0' shape=(None,) dtype=int64>, 'Parch': <tf.Tensor 'data_6:0' shape=(None,) dtype=int64>, 'Ticket': <tf.Tensor 'data_11:0' shape=(None,) dtype=string>, 'Fare': <t

[INFO 2024-02-11T15:37:17.542154173+00:00 kernel.cc:756] Start Yggdrasil model training
[INFO 2024-02-11T15:37:17.542204069+00:00 kernel.cc:757] Collect training examples
[INFO 2024-02-11T15:37:17.542533953+00:00 kernel.cc:388] Number of batches: 1
[INFO 2024-02-11T15:37:17.542552942+00:00 kernel.cc:389] Number of examples: 891
[INFO 2024-02-11T15:37:17.543869269+00:00 data_spec_inference.cc:303] 147 item(s) have been pruned (i.e. they are considered out of dictionary) for the column Cabin (0 item(s) left) because min_value_count=5 and max_number_of_unique_values=2000
[INFO 2024-02-11T15:37:17.544638637+00:00 data_spec_inference.cc:303] 1441 item(s) have been pruned (i.e. they are considered out of dictionary) for the column Name (85 item(s) left) because min_value_count=5 and max_number_of_unique_values=2000
[INFO 2024-02-11T15:37:17.544941959+00:00 data_spec_inference.cc:303] 28 item(s) have been pruned (i.e. they are considered out of dictionary) for the column Ticket_item (16 item(

Model trained in 0:00:01.080661
Compiling model...
Model compiled.
Accuracy: 0.739130437374115 Loss:1.0502052307128906


Let's look at the model and you can also notice the information about variable importance that the model figured out

In [8]:
model.summary()

Model: "gradient_boosted_trees_model_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
Total params: 1
Trainable params: 0
Non-trainable params: 1
_________________________________________________________________
Type: "GRADIENT_BOOSTED_TREES"
Task: CLASSIFICATION
Label: "__LABEL"

Input Features (11):
	Age
	Cabin
	Embarked
	Fare
	Name
	Parch
	Pclass
	Sex
	SibSp
	Ticket_item
	Ticket_number

No weights

Variable Importance: INV_MEAN_MIN_DEPTH:
    1.           "Sex"  0.611555 ################
    2.           "Age"  0.343173 ######
    3.          "Fare"  0.275439 ####
    4.          "Name"  0.189713 #
    5.        "Pclass"  0.172931 
    6.   "Ticket_item"  0.169311 
    7. "Ticket_number"  0.165249 
    8.         "Parch"  0.164456 
    9.      "Embarked"  0.163076 
   10.         "SibSp"  0.158163 

Variable Importance: MEAN_DECREASE_IN_ACCURACY:
    1.           "Sex"  0.141304 ################
  

# Make predictions

In [9]:
def prediction_to_kaggle_format(model, threshold=0.5):
    proba_survive = model.predict(serving_ds, verbose=0)[:,0]
    return pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (proba_survive >= threshold).astype(int)
    })

def make_submission(kaggle_predictions):
    path="/kaggle/working/submission.csv"
    kaggle_predictions.to_csv(path, index=False)
    print(f"Submission exported to {path}")
    
kaggle_predictions = prediction_to_kaggle_format(model)
make_submission(kaggle_predictions)
!head /kaggle/working/submission.csv

Submission exported to /kaggle/working/submission.csv
PassengerId,Survived
892,0
893,0
894,0
895,0
896,0
897,0
898,0
899,0
900,1


# Training a model with hyperparameter tunning

Hyper-parameter tuning is enabled by specifying the tuner constructor argument of the model. The tuner object contains all the configuration of the tuner (search space, optimizer, trial and objective).


In [10]:
tuner = tfdf.tuner.RandomSearch(num_trials=1000)
tuner.choice("min_examples", [2, 5, 7, 10])
tuner.choice("categorical_algorithm", ["CART", "RANDOM"])

local_search_space = tuner.choice("growing_strategy", ["LOCAL"])
local_search_space.choice("max_depth", [3, 4, 5, 6, 8])

global_search_space = tuner.choice("growing_strategy", ["BEST_FIRST_GLOBAL"], merge=True)
global_search_space.choice("max_num_nodes", [16, 32, 64, 128, 256])

#tuner.choice("use_hessian_gain", [True, False])
tuner.choice("shrinkage", [0.02, 0.05, 0.10, 0.15])
tuner.choice("num_candidate_attributes_ratio", [0.2, 0.5, 0.9, 1.0])


tuner.choice("split_axis", ["AXIS_ALIGNED"])
oblique_space = tuner.choice("split_axis", ["SPARSE_OBLIQUE"], merge=True)
oblique_space.choice("sparse_oblique_normalization",
                     ["NONE", "STANDARD_DEVIATION", "MIN_MAX"])
oblique_space.choice("sparse_oblique_weights", ["BINARY", "CONTINUOUS"])
oblique_space.choice("sparse_oblique_num_projections_exponent", [1.0, 1.5])

# Tune the model. Notice the `tuner=tuner`.
tuned_model = tfdf.keras.GradientBoostedTreesModel(tuner=tuner)
tuned_model.fit(train_ds, verbose=0)

tuned_self_evaluation = tuned_model.make_inspector().evaluation()
print(f"Accuracy: {tuned_self_evaluation.accuracy} Loss:{tuned_self_evaluation.loss}")

Use /tmp/tmpma4kmp0n as temporary training directory


[INFO 2024-02-11T15:39:24.866517238+00:00 kernel.cc:1214] Loading model from path /tmp/tmpma4kmp0n/model/ with prefix 866ad786969d4bcf
[INFO 2024-02-11T15:39:24.882952787+00:00 decision_forest.cc:661] Model loaded with 19 root(s), 589 node(s), and 12 input feature(s).
[INFO 2024-02-11T15:39:24.883180819+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-11T15:39:24.883275884+00:00 kernel.cc:1046] Use fast generic engine


Accuracy: 0.9178082346916199 Loss:0.6503586769104004


In the last line in the cell above, you can see the accuracy is higher than previously with default parameters and parameters set by hand.

This is the main idea behing hyperparameter tuning.

For more information you can follow this tutorial: [Automated hyper-parameter tuning](https://www.tensorflow.org/decision_forests/tutorials/automatic_tuning_colab)

# Making an ensemble

Here you'll create 100 models with different seeds and combine their results

This approach removes a little bit the random aspects related to creating ML models

In the GBT creation is used the `honest` parameter. It will use different training examples to infer the structure and the leaf values. This regularization technique trades examples for bias estimates.

In [11]:
predictions = None
num_predictions = 0

for i in range(100):
    print(f"i:{i}")
    # Possible models: GradientBoostedTreesModel or RandomForestModel
    model = tfdf.keras.GradientBoostedTreesModel(
        verbose=0, # Very few logs
        features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
        exclude_non_specified_features=True, # Only use the features in "features"

        #min_examples=1,
        #categorical_algorithm="RANDOM",
        ##max_depth=4,
        #shrinkage=0.05,
        ##num_candidate_attributes_ratio=0.2,
        #split_axis="SPARSE_OBLIQUE",
        #sparse_oblique_normalization="MIN_MAX",
        #sparse_oblique_num_projections_exponent=2.0,
        #num_trees=2000,
        ##validation_ratio=0.0,
        random_seed=i,
        honest=True,
    )
    model.fit(train_ds)
    
    sub_predictions = model.predict(serving_ds, verbose=0)[:,0]
    if predictions is None:
        predictions = sub_predictions
    else:
        predictions += sub_predictions
    num_predictions += 1

predictions/=num_predictions

kaggle_predictions = pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (predictions >= 0.5).astype(int)
    })

make_submission(kaggle_predictions)

i:0


[INFO 2024-02-11T15:39:26.069007646+00:00 kernel.cc:1214] Loading model from path /tmp/tmpqcs3xslq/model/ with prefix bf8bafb2d0194f17
[INFO 2024-02-11T15:39:26.073981825+00:00 kernel.cc:1046] Use fast generic engine


i:1


[INFO 2024-02-11T15:39:27.662971015+00:00 kernel.cc:1214] Loading model from path /tmp/tmpqa0xlf_3/model/ with prefix 145bb9b3fb7d4409
[INFO 2024-02-11T15:39:27.684182599+00:00 kernel.cc:1046] Use fast generic engine


i:2


[INFO 2024-02-11T15:39:28.795795034+00:00 kernel.cc:1214] Loading model from path /tmp/tmpoal2_xk9/model/ with prefix 6783cafbb4b942f7
[INFO 2024-02-11T15:39:28.80042415+00:00 kernel.cc:1046] Use fast generic engine


i:3


[INFO 2024-02-11T15:39:30.609290768+00:00 kernel.cc:1214] Loading model from path /tmp/tmpiywzq4ej/model/ with prefix b31b5b7ce01443df
[INFO 2024-02-11T15:39:30.637795312+00:00 kernel.cc:1046] Use fast generic engine


i:4


[INFO 2024-02-11T15:39:31.811194071+00:00 kernel.cc:1214] Loading model from path /tmp/tmpsztfiewc/model/ with prefix 2c7512b5db024836
[INFO 2024-02-11T15:39:31.817396416+00:00 kernel.cc:1046] Use fast generic engine


i:5


[INFO 2024-02-11T15:39:32.951511671+00:00 kernel.cc:1214] Loading model from path /tmp/tmph7r2do0p/model/ with prefix f960898f549a4019
[INFO 2024-02-11T15:39:32.955126456+00:00 kernel.cc:1046] Use fast generic engine


i:6


[INFO 2024-02-11T15:39:34.1632732+00:00 kernel.cc:1214] Loading model from path /tmp/tmpb3rygh3y/model/ with prefix 348fc0e891ab4438
[INFO 2024-02-11T15:39:34.171381428+00:00 kernel.cc:1046] Use fast generic engine


i:7


[INFO 2024-02-11T15:39:35.826586877+00:00 kernel.cc:1214] Loading model from path /tmp/tmp0caj16ap/model/ with prefix ca0b807cd47a44b9
[INFO 2024-02-11T15:39:35.84856761+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-11T15:39:35.848642312+00:00 kernel.cc:1046] Use fast generic engine


i:8


[INFO 2024-02-11T15:39:37.104787384+00:00 kernel.cc:1214] Loading model from path /tmp/tmpkzyf4gow/model/ with prefix 366775e081f4443a
[INFO 2024-02-11T15:39:37.115021151+00:00 kernel.cc:1046] Use fast generic engine


i:9


[INFO 2024-02-11T15:39:39.019333972+00:00 kernel.cc:1214] Loading model from path /tmp/tmpfkm7tol2/model/ with prefix e39f3971fc2b4347
[INFO 2024-02-11T15:39:39.035441311+00:00 kernel.cc:1046] Use fast generic engine


i:10


[INFO 2024-02-11T15:39:40.496538304+00:00 kernel.cc:1214] Loading model from path /tmp/tmp19kybikq/model/ with prefix 146643e77e414d82
[INFO 2024-02-11T15:39:40.502910271+00:00 kernel.cc:1046] Use fast generic engine


i:11


[INFO 2024-02-11T15:39:42.610406151+00:00 kernel.cc:1214] Loading model from path /tmp/tmpnxbrvpuj/model/ with prefix b32006f3538a47fb
[INFO 2024-02-11T15:39:42.629298892+00:00 kernel.cc:1046] Use fast generic engine


i:12


[INFO 2024-02-11T15:39:43.877675019+00:00 kernel.cc:1214] Loading model from path /tmp/tmp9olk3ncd/model/ with prefix 9529dc1c0ee14354
[INFO 2024-02-11T15:39:43.884460209+00:00 kernel.cc:1046] Use fast generic engine


i:13


[INFO 2024-02-11T15:39:45.292520582+00:00 kernel.cc:1214] Loading model from path /tmp/tmpiji42eva/model/ with prefix 09a5e4fea3c148b5
[INFO 2024-02-11T15:39:45.304623939+00:00 kernel.cc:1046] Use fast generic engine


i:14


[INFO 2024-02-11T15:39:46.847296352+00:00 kernel.cc:1214] Loading model from path /tmp/tmp7o2t03r2/model/ with prefix 3a0d5d2335be42d0
[INFO 2024-02-11T15:39:46.853994554+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-11T15:39:46.854052218+00:00 kernel.cc:1046] Use fast generic engine


i:15


[INFO 2024-02-11T15:39:48.069152146+00:00 kernel.cc:1214] Loading model from path /tmp/tmpqqr5uyhj/model/ with prefix b7436ea015b2434c
[INFO 2024-02-11T15:39:48.077552156+00:00 kernel.cc:1046] Use fast generic engine


i:16


[INFO 2024-02-11T15:39:49.473044853+00:00 kernel.cc:1214] Loading model from path /tmp/tmpd8lfdr9l/model/ with prefix a3b2dd9559a54af6
[INFO 2024-02-11T15:39:49.487259107+00:00 kernel.cc:1046] Use fast generic engine


i:17


[INFO 2024-02-11T15:39:50.901247222+00:00 kernel.cc:1214] Loading model from path /tmp/tmp5err71w2/model/ with prefix ad41777078bd4105
[INFO 2024-02-11T15:39:50.915738108+00:00 kernel.cc:1046] Use fast generic engine


i:18


[INFO 2024-02-11T15:39:52.283643012+00:00 kernel.cc:1214] Loading model from path /tmp/tmpm5v52tg8/model/ with prefix f012e56cc26740a2
[INFO 2024-02-11T15:39:52.29678304+00:00 kernel.cc:1046] Use fast generic engine


i:19


[INFO 2024-02-11T15:39:53.901884028+00:00 kernel.cc:1214] Loading model from path /tmp/tmpy2acfmt3/model/ with prefix b6267022eb1a4191
[INFO 2024-02-11T15:39:53.921806543+00:00 kernel.cc:1046] Use fast generic engine


i:20


[INFO 2024-02-11T15:39:55.405406709+00:00 kernel.cc:1214] Loading model from path /tmp/tmpwrvya21g/model/ with prefix 02e9cc542bc047f8
[INFO 2024-02-11T15:39:55.422988772+00:00 kernel.cc:1046] Use fast generic engine


i:21


[INFO 2024-02-11T15:39:56.570044498+00:00 kernel.cc:1214] Loading model from path /tmp/tmpd1pjwup_/model/ with prefix f5a9b00449cc4239
[INFO 2024-02-11T15:39:56.575559275+00:00 kernel.cc:1046] Use fast generic engine


i:22


[INFO 2024-02-11T15:39:57.751737451+00:00 kernel.cc:1214] Loading model from path /tmp/tmp3ys671k2/model/ with prefix 6617ebdc138d483e
[INFO 2024-02-11T15:39:57.758235358+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-11T15:39:57.758300566+00:00 kernel.cc:1046] Use fast generic engine


i:23


[INFO 2024-02-11T15:39:59.029399607+00:00 kernel.cc:1214] Loading model from path /tmp/tmpq4tul_2u/model/ with prefix 0d8029b1586c4581
[INFO 2024-02-11T15:39:59.039536987+00:00 kernel.cc:1046] Use fast generic engine


i:24


[INFO 2024-02-11T15:40:00.244182077+00:00 kernel.cc:1214] Loading model from path /tmp/tmp39yei687/model/ with prefix e7720a78d0d149b4
[INFO 2024-02-11T15:40:00.250699996+00:00 kernel.cc:1046] Use fast generic engine


i:25


[INFO 2024-02-11T15:40:01.645517186+00:00 kernel.cc:1214] Loading model from path /tmp/tmp2r27nnvy/model/ with prefix bb8427ba6e5f46d2
[INFO 2024-02-11T15:40:01.658312334+00:00 kernel.cc:1046] Use fast generic engine


i:26


[INFO 2024-02-11T15:40:02.948241808+00:00 kernel.cc:1214] Loading model from path /tmp/tmpl_5939ym/model/ with prefix 2948772fe56d49e0
[INFO 2024-02-11T15:40:02.959130878+00:00 kernel.cc:1046] Use fast generic engine


i:27


[INFO 2024-02-11T15:40:04.653627103+00:00 kernel.cc:1214] Loading model from path /tmp/tmpn84hh9ie/model/ with prefix 6ee95b3aa3bd4004
[INFO 2024-02-11T15:40:04.6610177+00:00 kernel.cc:1046] Use fast generic engine


i:28


[INFO 2024-02-11T15:40:05.815892826+00:00 kernel.cc:1214] Loading model from path /tmp/tmp2ry97yuj/model/ with prefix ced5fec8ea694b1e
[INFO 2024-02-11T15:40:05.821787582+00:00 kernel.cc:1046] Use fast generic engine


i:29


[INFO 2024-02-11T15:40:07.247968319+00:00 kernel.cc:1214] Loading model from path /tmp/tmp1p9rt9uh/model/ with prefix 9bf7a36aca6b4fbb
[INFO 2024-02-11T15:40:07.262302724+00:00 kernel.cc:1046] Use fast generic engine


i:30


[INFO 2024-02-11T15:40:09.093396563+00:00 kernel.cc:1214] Loading model from path /tmp/tmpfq0gl8y4/model/ with prefix 8e655a68cadc4883
[INFO 2024-02-11T15:40:09.12308234+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-11T15:40:09.123163858+00:00 kernel.cc:1046] Use fast generic engine


i:31


[INFO 2024-02-11T15:40:10.405094129+00:00 kernel.cc:1214] Loading model from path /tmp/tmp8b3tt1rq/model/ with prefix 1c30641072b14417
[INFO 2024-02-11T15:40:10.415611535+00:00 kernel.cc:1046] Use fast generic engine


i:32


[INFO 2024-02-11T15:40:11.594118794+00:00 kernel.cc:1214] Loading model from path /tmp/tmpi7gcf38q/model/ with prefix f4fa994d55cd41bf
[INFO 2024-02-11T15:40:11.600290739+00:00 kernel.cc:1046] Use fast generic engine


i:33


[INFO 2024-02-11T15:40:13.016287394+00:00 kernel.cc:1214] Loading model from path /tmp/tmpm9hz3mdb/model/ with prefix c661b4b6f32f4604
[INFO 2024-02-11T15:40:13.030407706+00:00 kernel.cc:1046] Use fast generic engine


i:34


[INFO 2024-02-11T15:40:15.209065095+00:00 kernel.cc:1214] Loading model from path /tmp/tmpzybe8l6s/model/ with prefix 1146d54412164e74
[INFO 2024-02-11T15:40:15.220766148+00:00 kernel.cc:1046] Use fast generic engine


i:35


[INFO 2024-02-11T15:40:16.543766426+00:00 kernel.cc:1214] Loading model from path /tmp/tmpj9m0d9mf/model/ with prefix 4ae4cd1ae3014e62
[INFO 2024-02-11T15:40:16.553199812+00:00 kernel.cc:1046] Use fast generic engine


i:36


[INFO 2024-02-11T15:40:18.049716614+00:00 kernel.cc:1214] Loading model from path /tmp/tmp9qyrvdk0/model/ with prefix ae7cc18ff6b641e1
[INFO 2024-02-11T15:40:18.066156586+00:00 kernel.cc:1046] Use fast generic engine


i:37


[INFO 2024-02-11T15:40:19.364240608+00:00 kernel.cc:1214] Loading model from path /tmp/tmpsx393c0i/model/ with prefix 8369bf2c7968423e
[INFO 2024-02-11T15:40:19.374119372+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-11T15:40:19.374186693+00:00 kernel.cc:1046] Use fast generic engine


i:38


[INFO 2024-02-11T15:40:20.766171339+00:00 kernel.cc:1214] Loading model from path /tmp/tmpglqirf9h/model/ with prefix 60e35d7193b64f65
[INFO 2024-02-11T15:40:20.78068494+00:00 kernel.cc:1046] Use fast generic engine


i:39


[INFO 2024-02-11T15:40:22.196276377+00:00 kernel.cc:1214] Loading model from path /tmp/tmped6cvv15/model/ with prefix e9989403369c4c64
[INFO 2024-02-11T15:40:22.210502645+00:00 kernel.cc:1046] Use fast generic engine


i:40


[INFO 2024-02-11T15:40:23.351077086+00:00 kernel.cc:1214] Loading model from path /tmp/tmppe2pnfke/model/ with prefix 3578f293b6194650
[INFO 2024-02-11T15:40:23.356387264+00:00 kernel.cc:1046] Use fast generic engine


i:41


[INFO 2024-02-11T15:40:24.85961336+00:00 kernel.cc:1214] Loading model from path /tmp/tmpsc__5a44/model/ with prefix 83f0e5d8fdf54fea
[INFO 2024-02-11T15:40:24.877149229+00:00 kernel.cc:1046] Use fast generic engine


i:42


[INFO 2024-02-11T15:40:26.415619612+00:00 kernel.cc:1214] Loading model from path /tmp/tmpzgt3_3j8/model/ with prefix cd1e9c6865d1417a
[INFO 2024-02-11T15:40:26.425885752+00:00 kernel.cc:1046] Use fast generic engine


i:43


[INFO 2024-02-11T15:40:27.993133643+00:00 kernel.cc:1214] Loading model from path /tmp/tmpu_iyt0uj/model/ with prefix d33901240daa472f
[INFO 2024-02-11T15:40:28.01200931+00:00 kernel.cc:1046] Use fast generic engine


i:44


[INFO 2024-02-11T15:40:29.299321112+00:00 kernel.cc:1214] Loading model from path /tmp/tmph0e27bwv/model/ with prefix 991eaae0b646489b
[INFO 2024-02-11T15:40:29.309652179+00:00 kernel.cc:1046] Use fast generic engine


i:45


[INFO 2024-02-11T15:40:30.883349072+00:00 kernel.cc:1214] Loading model from path /tmp/tmp5jto24ct/model/ with prefix b4b613fc8e3f48e1
[INFO 2024-02-11T15:40:30.887197448+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-11T15:40:30.887245319+00:00 kernel.cc:1046] Use fast generic engine


i:46


[INFO 2024-02-11T15:40:32.352464974+00:00 kernel.cc:1214] Loading model from path /tmp/tmphmkawmzi/model/ with prefix 9a493b902ee54e2b
[INFO 2024-02-11T15:40:32.368325599+00:00 kernel.cc:1046] Use fast generic engine


i:47


[INFO 2024-02-11T15:40:33.820641868+00:00 kernel.cc:1214] Loading model from path /tmp/tmpvpfvox9l/model/ with prefix c3ff69b7ec5d457c
[INFO 2024-02-11T15:40:33.834645727+00:00 kernel.cc:1046] Use fast generic engine


i:48


[INFO 2024-02-11T15:40:35.101419416+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_8jgyvwr/model/ with prefix 004f1694399043d1
[INFO 2024-02-11T15:40:35.106663665+00:00 kernel.cc:1046] Use fast generic engine


i:49


[INFO 2024-02-11T15:40:36.342772106+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_ivlhwi5/model/ with prefix d3d87d090659435c
[INFO 2024-02-11T15:40:36.350964857+00:00 kernel.cc:1046] Use fast generic engine


i:50


[INFO 2024-02-11T15:40:37.745537849+00:00 kernel.cc:1214] Loading model from path /tmp/tmpf_32sh0c/model/ with prefix b22812497bea45b0
[INFO 2024-02-11T15:40:37.758809626+00:00 kernel.cc:1046] Use fast generic engine


i:51


[INFO 2024-02-11T15:40:39.644852923+00:00 kernel.cc:1214] Loading model from path /tmp/tmpqcytw9hw/model/ with prefix b11aa1db53054ef2
[INFO 2024-02-11T15:40:39.661566983+00:00 kernel.cc:1046] Use fast generic engine


i:52


[INFO 2024-02-11T15:40:40.926882459+00:00 kernel.cc:1214] Loading model from path /tmp/tmph1mo991f/model/ with prefix 1ccde45989ee44bc
[INFO 2024-02-11T15:40:40.93525309+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-11T15:40:40.935297725+00:00 kernel.cc:1046] Use fast generic engine


i:53


[INFO 2024-02-11T15:40:42.186526768+00:00 kernel.cc:1214] Loading model from path /tmp/tmpfqzn480g/model/ with prefix 22c9974ae9554e6a
[INFO 2024-02-11T15:40:42.195119153+00:00 kernel.cc:1046] Use fast generic engine


i:54


[INFO 2024-02-11T15:40:43.835155392+00:00 kernel.cc:1214] Loading model from path /tmp/tmpt6q71gh3/model/ with prefix 2df04d5309fb4528
[INFO 2024-02-11T15:40:43.83881725+00:00 kernel.cc:1046] Use fast generic engine


i:55


[INFO 2024-02-11T15:40:45.337418142+00:00 kernel.cc:1214] Loading model from path /tmp/tmp0k_ne7b0/model/ with prefix c32a0ca451bb4ec7
[INFO 2024-02-11T15:40:45.354283381+00:00 kernel.cc:1046] Use fast generic engine


i:56


[INFO 2024-02-11T15:40:46.734482402+00:00 kernel.cc:1214] Loading model from path /tmp/tmp11o2afdc/model/ with prefix 030b410bfc0a4c37
[INFO 2024-02-11T15:40:46.746200992+00:00 kernel.cc:1046] Use fast generic engine


i:57


[INFO 2024-02-11T15:40:47.879864479+00:00 kernel.cc:1214] Loading model from path /tmp/tmpuu2gmny0/model/ with prefix 823bec62ba5d4bee
[INFO 2024-02-11T15:40:47.885916077+00:00 kernel.cc:1046] Use fast generic engine


i:58


[INFO 2024-02-11T15:40:49.696668067+00:00 kernel.cc:1214] Loading model from path /tmp/tmp2y4sssr1/model/ with prefix b859059ed1db42ba
[INFO 2024-02-11T15:40:49.703820939+00:00 kernel.cc:1046] Use fast generic engine


i:59


[INFO 2024-02-11T15:40:51.572461439+00:00 kernel.cc:1214] Loading model from path /tmp/tmpd57xu425/model/ with prefix fc3d45fa81564659
[INFO 2024-02-11T15:40:51.582844245+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-11T15:40:51.582895726+00:00 kernel.cc:1046] Use fast generic engine


i:60


[INFO 2024-02-11T15:40:54.03381381+00:00 kernel.cc:1214] Loading model from path /tmp/tmphtfxx2ns/model/ with prefix 5338182995d84210
[INFO 2024-02-11T15:40:54.046473527+00:00 kernel.cc:1046] Use fast generic engine


i:61


[INFO 2024-02-11T15:40:55.356550181+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_4g7mqcj/model/ with prefix 273811729d484b5a
[INFO 2024-02-11T15:40:55.362501305+00:00 kernel.cc:1046] Use fast generic engine


i:62


[INFO 2024-02-11T15:40:57.29654629+00:00 kernel.cc:1214] Loading model from path /tmp/tmpoida6djb/model/ with prefix c9a7323efbe24337
[INFO 2024-02-11T15:40:57.327087185+00:00 kernel.cc:1046] Use fast generic engine


i:63


[INFO 2024-02-11T15:40:59.830848416+00:00 kernel.cc:1214] Loading model from path /tmp/tmpmueazmb7/model/ with prefix abdaa9c081fe4c84
[INFO 2024-02-11T15:40:59.841314087+00:00 kernel.cc:1046] Use fast generic engine


i:64


[INFO 2024-02-11T15:41:01.128039713+00:00 kernel.cc:1214] Loading model from path /tmp/tmpxxt3nn3m/model/ with prefix a0becaf0ef064a11
[INFO 2024-02-11T15:41:01.137041388+00:00 kernel.cc:1046] Use fast generic engine


i:65


[INFO 2024-02-11T15:41:02.323385336+00:00 kernel.cc:1214] Loading model from path /tmp/tmpbrh0nsno/model/ with prefix 9ce1338aec5543b2
[INFO 2024-02-11T15:41:02.32930335+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-11T15:41:02.329364802+00:00 kernel.cc:1046] Use fast generic engine


i:66


[INFO 2024-02-11T15:41:03.544387122+00:00 kernel.cc:1214] Loading model from path /tmp/tmpjfox9d4w/model/ with prefix ea56a44d6ca648f8
[INFO 2024-02-11T15:41:03.551677296+00:00 kernel.cc:1046] Use fast generic engine


i:67


[INFO 2024-02-11T15:41:05.21479499+00:00 kernel.cc:1214] Loading model from path /tmp/tmpmc5_q_bd/model/ with prefix 2519520d57f6443b
[INFO 2024-02-11T15:41:05.23553841+00:00 kernel.cc:1046] Use fast generic engine


i:68


[INFO 2024-02-11T15:41:06.73526948+00:00 kernel.cc:1214] Loading model from path /tmp/tmpuivrqdxa/model/ with prefix 73df991612e4420a
[INFO 2024-02-11T15:41:06.748634942+00:00 kernel.cc:1046] Use fast generic engine


i:69


[INFO 2024-02-11T15:41:08.259203797+00:00 kernel.cc:1214] Loading model from path /tmp/tmpvi7t6fel/model/ with prefix d3167e0584404da6
[INFO 2024-02-11T15:41:08.26639721+00:00 kernel.cc:1046] Use fast generic engine


i:70


[INFO 2024-02-11T15:41:09.552988372+00:00 kernel.cc:1214] Loading model from path /tmp/tmpiz43gq3_/model/ with prefix af447c7527074e00
[INFO 2024-02-11T15:41:09.563496588+00:00 kernel.cc:1046] Use fast generic engine


i:71


[INFO 2024-02-11T15:41:11.289729293+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_wtm5gf1/model/ with prefix e7cb3f7efc254830
[INFO 2024-02-11T15:41:11.298182098+00:00 kernel.cc:1046] Use fast generic engine


i:72


[INFO 2024-02-11T15:41:13.307625169+00:00 kernel.cc:1214] Loading model from path /tmp/tmptg4p_j_e/model/ with prefix 92b602410b374df6
[INFO 2024-02-11T15:41:13.324101027+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-11T15:41:13.324152896+00:00 kernel.cc:1046] Use fast generic engine


i:73


[INFO 2024-02-11T15:41:14.596142922+00:00 kernel.cc:1214] Loading model from path /tmp/tmpcqxjrm4j/model/ with prefix 7c532f6886264920
[INFO 2024-02-11T15:41:14.604174232+00:00 kernel.cc:1046] Use fast generic engine


i:74


[INFO 2024-02-11T15:41:16.015075229+00:00 kernel.cc:1214] Loading model from path /tmp/tmp6cf2oycv/model/ with prefix d87ac26cdfe940e7
[INFO 2024-02-11T15:41:16.028283824+00:00 kernel.cc:1046] Use fast generic engine


i:75


[INFO 2024-02-11T15:41:17.388368924+00:00 kernel.cc:1214] Loading model from path /tmp/tmpsohqtouh/model/ with prefix 00e4342966ad4ba4
[INFO 2024-02-11T15:41:17.397503901+00:00 kernel.cc:1046] Use fast generic engine


i:76


[INFO 2024-02-11T15:41:18.577949379+00:00 kernel.cc:1214] Loading model from path /tmp/tmpezt_l7dc/model/ with prefix 55f92d6a5701440f
[INFO 2024-02-11T15:41:18.582306097+00:00 kernel.cc:1046] Use fast generic engine


i:77


[INFO 2024-02-11T15:41:19.731131868+00:00 kernel.cc:1214] Loading model from path /tmp/tmpzd2120_o/model/ with prefix 3c6766c261634606
[INFO 2024-02-11T15:41:19.736253764+00:00 kernel.cc:1046] Use fast generic engine


i:78


[INFO 2024-02-11T15:41:20.966215311+00:00 kernel.cc:1214] Loading model from path /tmp/tmpllhr7t21/model/ with prefix ecdb5f531046473e
[INFO 2024-02-11T15:41:20.973682504+00:00 kernel.cc:1046] Use fast generic engine


i:79


[INFO 2024-02-11T15:41:22.791501489+00:00 kernel.cc:1214] Loading model from path /tmp/tmpy1h3p5lh/model/ with prefix 62751ec184a547a0
[INFO 2024-02-11T15:41:22.799391423+00:00 kernel.cc:1046] Use fast generic engine


i:80


[INFO 2024-02-11T15:41:24.15928139+00:00 kernel.cc:1214] Loading model from path /tmp/tmpxxowpi3y/model/ with prefix cf7bb87ba8694741
[INFO 2024-02-11T15:41:24.169981172+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-11T15:41:24.17002794+00:00 kernel.cc:1046] Use fast generic engine


i:81


[INFO 2024-02-11T15:41:25.981020708+00:00 kernel.cc:1214] Loading model from path /tmp/tmpq6o3eyrn/model/ with prefix e8094652ef264350
[INFO 2024-02-11T15:41:25.993147313+00:00 kernel.cc:1046] Use fast generic engine


i:82


[INFO 2024-02-11T15:41:27.319808989+00:00 kernel.cc:1214] Loading model from path /tmp/tmpfxtxa6_5/model/ with prefix 9d5dfd74d7df4cc8
[INFO 2024-02-11T15:41:27.330210711+00:00 kernel.cc:1046] Use fast generic engine


i:83


[INFO 2024-02-11T15:41:29.081757968+00:00 kernel.cc:1214] Loading model from path /tmp/tmpz5eur_nw/model/ with prefix c0ee4a4defb54ab3
[INFO 2024-02-11T15:41:29.091804166+00:00 kernel.cc:1046] Use fast generic engine


i:84


[INFO 2024-02-11T15:41:30.625146872+00:00 kernel.cc:1214] Loading model from path /tmp/tmpc_zxalmg/model/ with prefix c4c69f2748e84302
[INFO 2024-02-11T15:41:30.642643005+00:00 kernel.cc:1046] Use fast generic engine


i:85


[INFO 2024-02-11T15:41:31.857587465+00:00 kernel.cc:1214] Loading model from path /tmp/tmpy0oi13bw/model/ with prefix ca46731373a54194
[INFO 2024-02-11T15:41:31.864924645+00:00 kernel.cc:1046] Use fast generic engine


i:86


[INFO 2024-02-11T15:41:33.419298822+00:00 kernel.cc:1214] Loading model from path /tmp/tmpzdh13hi_/model/ with prefix 7549352d84564025
[INFO 2024-02-11T15:41:33.436557012+00:00 kernel.cc:1046] Use fast generic engine


i:87


[INFO 2024-02-11T15:41:36.117623752+00:00 kernel.cc:1214] Loading model from path /tmp/tmpaqw1wd97/model/ with prefix 1ea752a13e1f40a1
[INFO 2024-02-11T15:41:36.145587113+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-11T15:41:36.145662139+00:00 kernel.cc:1046] Use fast generic engine


i:88


[INFO 2024-02-11T15:41:37.691333211+00:00 kernel.cc:1214] Loading model from path /tmp/tmp5tvdly8x/model/ with prefix fcbf1c48bbaa4d3a
[INFO 2024-02-11T15:41:37.707504791+00:00 kernel.cc:1046] Use fast generic engine


i:89


[INFO 2024-02-11T15:41:39.338869232+00:00 kernel.cc:1214] Loading model from path /tmp/tmpf_pqv8gu/model/ with prefix 58f062fc5d6e45a4
[INFO 2024-02-11T15:41:39.345090379+00:00 kernel.cc:1046] Use fast generic engine


i:90


[INFO 2024-02-11T15:41:41.313295868+00:00 kernel.cc:1214] Loading model from path /tmp/tmp8uk8vfvz/model/ with prefix 45a538c05d494a2f
[INFO 2024-02-11T15:41:41.325429628+00:00 kernel.cc:1046] Use fast generic engine


i:91


[INFO 2024-02-11T15:41:42.978062134+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_r2gk98g/model/ with prefix ffd68505623046e8
[INFO 2024-02-11T15:41:42.985975949+00:00 kernel.cc:1046] Use fast generic engine


i:92


[INFO 2024-02-11T15:41:44.651338362+00:00 kernel.cc:1214] Loading model from path /tmp/tmpyhag4bkt/model/ with prefix dbfb2fedfde84517
[INFO 2024-02-11T15:41:44.671305671+00:00 kernel.cc:1046] Use fast generic engine


i:93


[INFO 2024-02-11T15:41:46.043036517+00:00 kernel.cc:1214] Loading model from path /tmp/tmpenjevhze/model/ with prefix 5d49f6444a5e4c63
[INFO 2024-02-11T15:41:46.054301392+00:00 kernel.cc:1046] Use fast generic engine


i:94


[INFO 2024-02-11T15:41:47.293810975+00:00 kernel.cc:1214] Loading model from path /tmp/tmp5n6u4l1p/model/ with prefix 196727f0ae0c4612
[INFO 2024-02-11T15:41:47.301435875+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-11T15:41:47.301496939+00:00 kernel.cc:1046] Use fast generic engine


i:95


[INFO 2024-02-11T15:41:48.661096205+00:00 kernel.cc:1214] Loading model from path /tmp/tmp636_dfqo/model/ with prefix d56d9607e04d491e
[INFO 2024-02-11T15:41:48.670367444+00:00 kernel.cc:1046] Use fast generic engine


i:96


[INFO 2024-02-11T15:41:50.491327693+00:00 kernel.cc:1214] Loading model from path /tmp/tmpcbqvt433/model/ with prefix f9a08ab10f2645d3
[INFO 2024-02-11T15:41:50.502276262+00:00 kernel.cc:1046] Use fast generic engine


i:97


[INFO 2024-02-11T15:41:51.657178439+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgbu8ytwo/model/ with prefix f63bf4825a234a1e
[INFO 2024-02-11T15:41:51.662890401+00:00 kernel.cc:1046] Use fast generic engine


i:98


[INFO 2024-02-11T15:41:52.919322852+00:00 kernel.cc:1214] Loading model from path /tmp/tmphg48jt03/model/ with prefix 3b840570ceaf48ef
[INFO 2024-02-11T15:41:52.927944506+00:00 kernel.cc:1046] Use fast generic engine


i:99


[INFO 2024-02-11T15:41:54.482689544+00:00 kernel.cc:1214] Loading model from path /tmp/tmpufvqcxxv/model/ with prefix 52ff2b47e8344201
[INFO 2024-02-11T15:41:54.498360266+00:00 kernel.cc:1046] Use fast generic engine


Submission exported to /kaggle/working/submission.csv


# What is next

If you want to learn more about TensorFlow Decision Forests and its advanced features, you can follow the official documentation [here](https://www.tensorflow.org/decision_forests) 